File: sportsbook-app/betting-service/src/main/java/com/sportsbook/bettingservice/outbox/OutboxUtil.java
package com.sportsbook.bettingservice.outbox;

import org.apache.avro.io.EncoderFactory;
import org.apache.avro.specific.SpecificDatumWriter;
import org.apache.avro.specific.SpecificRecord;

import java.io.ByteArrayOutputStream;

public class OutboxUtil {
    public static <T extends SpecificRecord> byte[] avroBytes(T record) throws Exception {
        var out = new ByteArrayOutputStream();
        var w = new SpecificDatumWriter<T>(record.getSchema());
        var enc = EncoderFactory.get().binaryEncoder(out, null);
        w.write(record, enc);
        enc.flush();
        return out.toByteArray();
    }
}

File: sportsbook-app/betting-service/src/main/kotlin/com/sportsbook/bettingservice/outbox/OutboxUtil.kt
package com.sportsbook.bettingservice.outbox

import org.apache.avro.io.EncoderFactory
import org.apache.avro.specific.SpecificDatumWriter
import org.apache.avro.specific.SpecificRecord
import java.io.ByteArrayOutputStream

object OutboxUtil {
    @Throws(Exception::class)
    fun <T : SpecificRecord> avroBytes(record: T): ByteArray {
        val out = ByteArrayOutputStream()
        val writer = SpecificDatumWriter<T>(record.schema)
        val enc = EncoderFactory.get().binaryEncoder(out, null)
        writer.write(record, enc)
        enc.flush()
        return out.toByteArray()
    }
}
